File: Haskell/006.hs
main = print no006

no006 :: Int
no006 =  squareOfSum - sumOfSquares

sumOfSquares :: Int
sumOfSquares = truncate $ sum [x**2 | x <- [1..100]]

squareOfSum :: Int
squareOfSum = truncate $ (sum [1..100]) ** 2

File: Elixir/006.exs
defmodule Math do
  def sumOfSquares do
    Enum.map(1..100, &(&1 * &1))
      |> Enum.sum
  end

  def squareOfSum do
    x = Enum.sum(1..100)
    x * x
  end
end

IO.puts abs(Math.sumOfSquares - Math.squareOfSum)
